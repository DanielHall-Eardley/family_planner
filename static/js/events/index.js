import createBtn from '../common/createBtn.js'
import postRequestAuth from '../common/postRequestAuth.js'
import inputListener from '../common/inputListener'
import addFormListeners from '../common/addFormListeners'
import eventForm from './eventForm.js'

import { FormState } from '../common/state'
export const formState = new FormState()

/*
Create an event form with JavaScript,
add input and form submission event listeners
*/
export const createEvent = () => {
  const main = document.querySelector('main')
  const form = document.createElement('form')

  form.className = 'create-event'
  form.addEventListener('submit', async (event) => {
    event.preventDefault()
    const body = formState.getCurrentState()
    const url = '/events/create'
    const token = localStorage.getItem('token')
    const response = await postRequestAuth(body, url, token)
    console.log(response)
  })

  form.appendChild(eventForm)
  addFormListeners(form, inputListener, formState)
  main.appendChild(form)
}

/*Attach the form creation to a button*/
window.onload = () => {
  createBtn(createEvent, document)
}
