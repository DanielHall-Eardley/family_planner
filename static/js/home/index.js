import state from '../common/state.js'

console.log(state.toDoList)