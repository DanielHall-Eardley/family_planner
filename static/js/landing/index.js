import checkExistingUser from './checkExistingUser.js'

window.onload = async () => {
  checkExistingUser()
}