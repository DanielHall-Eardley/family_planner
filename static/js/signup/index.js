import inputListener from '../common/inputListener'
import addFormListeners from '../common/addFormListeners'
import formCb from  './formCb'

import { FormState } from '../common/state'
export const formState = new FormState()

window.onload = () => {
  const form = document.querySelector('.signup-form')
  addFormListeners(form, inputListener, formState)

  form.addEventListener('submit', formCb)
}
