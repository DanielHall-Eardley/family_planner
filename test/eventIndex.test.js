import { createEvent } from '../static/js/events/index.js'

beforeAll(() => {
  const main = document.createElement('main')
  document.body.appendChild(main)
})

test('Event creation form matches snapshot', () => {
  createEvent()

  expect(document.querySelector('main')).toMatchSnapshot()
}) 
