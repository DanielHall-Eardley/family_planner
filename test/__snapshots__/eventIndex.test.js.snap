// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event creation form matches snapshot 1`] = `
<main>
  <form
    class="create-event"
  >
    <h3
      class="form-title"
    >
      Create Event
    </h3>
    <label
      for="alert-title"
    >
      Enter a name for your event
    </label>
    <input
      id="alert-title"
      name="title"
      type="text"
    />
    <label
      for="alert-date"
    >
      Enter a date for your event
    </label>
    <input
      id="alert-date"
      name="date"
      type="date"
    />
    <label
      for="img-url"
    >
      Link an image to your event
    </label>
    <input
      id="img-url"
      name="imgUrl"
      type="text"
    />
    <label
      for="alert-type"
    >
      Enter a brief description
    </label>
    <select
      class="alert-type"
      name="eventType"
    >
      <option
        disabled="true"
        selected="true"
      >
        Select a event type
      </option>
      <option>
        Boring chore task
      </option>
      <option>
        Happy social funtimes
      </option>
    </select>
    <label
      for="alert-desc"
    >
      Enter a brief description
    </label>
    <textarea
      id="alert-desc"
      name="eventDescription"
    />
    <button
      class="form-submit"
    >
      Create
    </button>
  </form>
</main>
`;
